File: src/main/java/com/github/zxh/classpy/classfile/constant/ConstantIntegerInfo.java
package com.github.zxh.classpy.classfile.constant;

/*
CONSTANT_Integer_info {
    u1 tag;
    u4 bytes;
}
*/
public class ConstantIntegerInfo extends ConstantInfo {

    {
        u4("bytes");
    }

    @Override
    protected String loadDesc(ConstantPool cp) {
        int i = super.getUInt("bytes");
        return String.valueOf(i);
    }
    
}

File: src/main/java/com/github/zxh/classpy/classfile/constant/ConstantIntegerInfo.kt
package com.github.zxh.classpy.classfile.constant;

/*
CONSTANT_Integer_info {
    u1 tag;
    u4 bytes;
}
*/
class ConstantIntegerInfo : ConstantInfo() {

    init {
        u4("bytes");
    }

    override fun loadDesc(cp: ConstantPool): String {
        val i = super.getInt("bytes");
        return i.toString();
    }
    
}
